packages/lib-utils/src/k8s/k8s-utils.ts

```diff
@@ -50,3 +62,189 @@ export const getK8sResourceURL = (
+    path: '',
+    host: 'auto',
+    reconnect: true,
+    jsonParse: true,
+    bufferFlushInterval: 500,
+    bufferMax: 1000,
+    ...wsOptions,
+  };
+
+  const { labelSelector } = query;
+  if (labelSelector) {
+    const encodedSelector = encodeURIComponent(selectorToString(labelSelector));
+    if (encodedSelector) {
+      queryParams.labelSelector = encodedSelector;
+    }
+  }
+
+  if (query.fieldSelector) {
+    queryParams.fieldSelector = encodeURIComponent(query.fieldSelector);
+  }
+
+  if (query.ns) {
+    opts.ns = query.ns;
+  }
+
+  if (query.resourceVersion) {
+    queryParams.resourceVersion = encodeURIComponent(query.resourceVersion);
+  }
+
+  const path = getK8sResourceURL(kind, undefined, opts);
+  wsOptionsUpdated.path = path;
+
+  // TODO utilize ws-factory
+  return { path, wsOptionsUpdated };
+  // return new WSFactory(path, wsOptionsUpdated as WSOptions);
+};
+
+/**
+ * Provides a group, version, and kind for a k8s model.
+ * @param model k8s model
+ * @returns The group, version, kind for the provided model.
+ * If the model does not have an apiGroup, group "core" will be returned.
+ * */
+export const getGroupVersionKindForModel: GetGroupVersionKindForModel = ({
+  apiGroup,
+  apiVersion,
+  kind,
+}) => ({
+  ...(apiGroup && { group: apiGroup }),
+  apiVersion,
+  kind,
+});
+
+/**
+ * @deprecated - This will become obsolete when we move away from K8sResourceKindReference to K8sGroupVersionKind
+ * Provides a reference string that uniquely identifies the group, version, and kind of a k8s resource.
+ * @param K8sGroupVersionKind Pass K8sGroupVersionKind which will have group, version, and kind of a k8s resource.
+ * @param K8sGroupVersionKind.apiGroup Pass group of k8s resource or model.
+ * @param K8sGroupVersionKind.apiVersion Pass version of k8s resource or model.
+ * @param K8sGroupVersionKind.kind Pass kind of k8s resource or model.
+ * @return The reference for any k8s resource i.e `group~version~kind`.
+ * If the group will not be present then "core" will be returned as part of the group in reference.
+ * * */
+export const getReference = ({
+  apiGroup,
+  apiVersion,
+  kind,
+}: K8sResourceIdentifier): K8sResourceKindReference =>
+  [apiGroup || 'core', apiVersion, kind].join('~');
+
+/**
+ * @deprecated - This will become obsolete when we move away from K8sResourceKindReference to K8sGroupVersionKind
+ * @see getGroupVersionKindForModel
+ * Provides a reference string that uniquely identifies the group, version, and kind of a k8s model.
+ * @param model k8s model
+ * @return The reference for model i.e `group~version~kind`.
+ * * */
+export const getReferenceForModel = (model: K8sModelCommon): K8sResourceKindReference =>
+  getReference({ apiGroup: model.apiGroup, apiVersion: model.apiVersion, kind: model.kind });
+
+// TODO Migrate implementation or refactor K8s reducer to avoid usage
+let k8sModels: ImmutableMap<K8sResourceKindReference, K8sModelCommon>;
+
+export const allModels = () => {
```

Comment:
@andrewballantyne @vojtechszocs this logic (not yet migrated) ultimately pulls in the console list of static models. See https://github.com/openshift/console/blob/master/frontend/packages/console-dynamic-plugin-sdk/src/utils/k8s/k8s-utils.ts#L215 and https://github.com/openshift/console/blob/master/frontend/public/models/index.ts

This PR is already pretty large, so I opted to stub it out for now until we review what I have already.

Reply:
Additionally, the missing logic requires changes to our `Extension` types in the core package.

Reply:
We should _need_ the models... they are additional details of custom nature (colour for instance)... this is fine to be omitted on first pass. We'll no doubt have to iterate on everything.